### DISCUSSION TITLE
Hey there, not so much a question about the project but more about what's going on at the moment

### DISCUSSION THREAD

**Discussion by MathyFey** [godot-jolt/godot-jolt#838] [Questions]
The Godot team says in a few places that they want to implement Jolt as the default physics engine on Godot in the future.
You made a lot of work to get the gdextension version where it is right now, you think they'll somehow make the extension part of the engine or make integrate it directly into the source?

This is a bit of a pickle right because since they say they want to add it officially, it would be a bit awkward to work on new features right?

Are you thinking about this aswell?  

Anyway, I love your work with the engine o/ I'm asking because I'm actually curious I'm not trying to be mean or annoying.

Thanks o/

**Comment by mihe** [comment]
> you think they'll somehow make the extension part of the engine or make integrate it directly into the source?

The plan is to have it be integrated into the engine as an actual engine module.

> This is a bit of a pickle right because since they say they want to add it officially, it would be a bit awkward to work on new features right?
>
> Are you thinking about this aswell?

Yes, I've actively avoided adding anything that isn't already in the default physics interface for this exact reason, for a long time now. It's highly unlikely that this extension will see more features added to it, other than the stuff that's pending on the master branch right now, like soft bodies, double precision and internal edge detection.
